File: models/follower.py
#!/usr/bin/python3
"""Defines the Follower class for user following system"""
from sqlalchemy import Column, String, ForeignKey
from models.base_model import BaseModel, Base
from sqlalchemy.orm import relationship

class Follower(Base):
    """Represents a user following another user"""
    __tablename__ = 'followers'

    follower_id = Column(
        String(60),
        ForeignKey('users.id', ondelete='CASCADE'),
        primary_key=True
    )
    followed_id = Column(
        String(60),
        ForeignKey('users.id', ondelete='CASCADE'),
        primary_key=True
    )

File: api/models/follower.go
package models

import (
	"github.com/google/uuid"
)

// Follower represents the user-following relationship.
type Follower struct {
	FollowerID uuid.UUID `gorm:"type:char(36);not null;primaryKey" json:"follower_id"`
	FollowedID uuid.UUID `gorm:"type:char(36);not null;primaryKey" json:"followed_id"`

	// Relationships
	Follower User `gorm:"foreignKey:FollowerID;constraint:OnDelete:CASCADE" json:"follower"`
	Followed User `gorm:"foreignKey:FollowedID;constraint:OnDelete:CASCADE" json:"followed"`
}

// TableName overrides the table name.
func (Follower) TableName() string {
	return "followers"
}
